app/Filament/Traits/HasParentResource.php

```diff
@@ -46,25 +48,22 @@ public function getParentRelationshipKey(): string
     {
         // You can set Custom relationship key (if it does not match the table name pattern) via $relationshipKey property.
         // Otherwise, it will be auto-resolved.
-        return $this->relationshipKey ?? str($this->parent?->getTable())->singular()->append('_id')->toString();
+        return $this->relationshipKey ?? $this->parent?->getForeignKey();
     }
 
     public function getBreadcrumbs(): array
     {
-        $resource = $this->getResource();
-        $parentResource = $this->getParentResource();
+        $resource = static::getResource();
+        $parentResource = static::getParentResource();
 
         $breadcrumbs = [
             $parentResource::getUrl() => $parentResource::getBreadCrumb(),
-            '#parent' => $parentResource::getRecordTitle($this->parent),
-
-            $parentResource::getUrl($resource::getPluralModelLabel() . '.index', [
-                'parent' => $this->parent,
-            ]) => $resource::getBreadCrumb(),
+            $parentResource::getRecordTitle($this->parent),
+            url($this->getName(), ['parent' => $this->parent]) => $resource::getBreadCrumb(),
         ];
 
         if (isset($this->record)) {
-            $breadcrumbs['#'] = $resource::getRecordTitle($this->record);
+            $breadcrumbs[] = $resource::getRecordTitle($this->record);
```

Comment:
Filament breadcrumbs array doesn't need a key, it will default to `#` if its not provided

Reply:
Heh, this is probably one change that I can't really agree with. Something here is messing up the URL pathing (I suspect the `url($this->getName()....)` part as it produces double APP URL on the breadcrumbs:
![image](https://github.com/LaravelDaily/filament-nested-resources/assets/3809773/d0f04ca2-b9ce-494c-b6a8-0ca0d0f14ded)